From bdae998b460e62beac0bf0c801dfcf5e8224dc5c Mon Sep 17 00:00:00 2001
From: Wang ShaoBo <bobo.shaobowang@huawei.com>
Date: Fri, 31 Dec 2021 15:59:45 +0800
Subject: [PATCH] kabi: reserve space for struct module
Patch-mainline: Not yet, from openEuler
References: bsn#22
openEuler-commit: bdae998b460e62beac0bf0c801dfcf5e8224dc5c


hulk inclusion
category: feature
bugzilla: https://gitee.com/openeuler/kernel/issues/I4KFY7?from=project-issue
CVE: NA

-------------------------------

Reserve space for struct module.

Signed-off-by: Wang ShaoBo <bobo.shaobowang@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Reviewed-by: Cheng Jian <cj.chengjian@huawei.com>
Signed-off-by: Zheng Zengkai <zhengzengkai@huawei.com>
Signed-off-by: Guoqing Jiang <guoqing.jiang@suse.com>
---
 include/linux/module.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/include/linux/module.h b/include/linux/module.h
index d7c6792d705b..54cdd20fc3de 100644
--- a/include/linux/module.h
+++ b/include/linux/module.h
@@ -553,6 +553,10 @@ struct module {
 	struct error_injection_entry *ei_funcs;
 	unsigned int num_ei_funcs;
 #endif
+	KABI_RESERVE(1)
+	KABI_RESERVE(2)
+	KABI_RESERVE(3)
+	KABI_RESERVE(4)
 } ____cacheline_aligned __randomize_layout;
 #ifndef MODULE_ARCH_INIT
 #define MODULE_ARCH_INIT {}
-- 
2.26.2

